### DISCUSSION TITLE
How to keep track of an untitled text document after saving it to disk

### DISCUSSION THREAD

**Discussion by EFanZh** [microsoft/vscode-discussions#475] [Extension Development QnA] [answered]
After saving an untitled text document, the original untitled `TextDocument` object is closed, and a new `TextDocument` is opened, the question is how do I know which newly opened `TextDocument` corresponds to the one that is just closed? They are not the same object, and the have different document URIs.

The case is that I want to associate a document with a webview to preview its content, but after saving an untitled document, the old document object is closed, so I lost the association between the document and the webview. Even though I can listen to the `onDidOpenTextDocument` event, I have no way of knowing whether I should associate the newly opened document to the existing webview.

Is there anyway to restore the association after saving an untitled document?

**Comment by alefragnani** [comment, marked as answer]
Hi @EFanZh ,

I suppose there is no way to do that, as I have the feeling your request is quite similar to the one I made almost an year ago, for `onWill/DidSaveAsTextDocument` events (https://github.com/microsoft/vscode/issues/146864). 

Back then I needed to know the _previous file_ in order to migrate data associated from one to another, but no API was available and `onDidRenameFiles` weren’t fired. In the end, unfortunately, the feature request didn’t achieve the required number of upvotes to enter the backlog, and it was closed.

Maybe, because you deal with _untitled_ files, something special is available for you to track it, but I think there isn’t. At least, not yet.

Hope this helps.
